src/dnsmasq_interface.c

```diff
@@ -2278,6 +2278,12 @@ void FTL_duplicate_reply(const int id, int *firstID)
 	duplicated_query->dnssec = source_query->dnssec;
 	duplicated_query->flags.complete = true;
 
+	// The original query may have been blocked during CNAME inspection,
+	// correct status in this case
+	if(duplicated_query->status != QUERY_FORWARDED)
```

Comment:
If I understand this correctly `source_query` is the query that actually got forwarded and `duplicated_query` is the query that was "ignored" for the time being. Should this then not be `if(source_query->status != QUERY_FORWARDED)` since the "ignored" query's status will never be `QUERY_FORWARDED`?

Reply:
Yes.